CMakeLists.txt

```diff
@@ -1,30 +1,43 @@
 cmake_minimum_required(VERSION 2.8.3)
 project(lms1xx)
 
+add_compile_options(-std=c++11)
+
 # Build ROS-independent library.
 find_package(console_bridge REQUIRED)
-include_directories(include ${console_bridge_INCLUDE_DIRS})
+include_directories(include SYSTEM ${console_bridge_INCLUDE_DIRS})
+
+# CoLaA Library abstracting protocol and implementing LMS1xx communication
+add_library(CoLaA src/colaa.cpp src/parse_helpers.cpp)
+target_link_libraries(CoLaA ${console_bridge_LIBRARIES})
 
-add_library(LMS1xx src/LMS1xx.cpp)
-target_link_libraries(LMS1xx ${console_bridge_LIBRARIES})
+# Specialisations for LMS5xx series scanners
+add_library(LMS5xx src/lms5xx.cpp)
+target_link_libraries(LMS5xx CoLaA ${console_bridge_LIBRARIES})
 
 add_library(MRS1000 src/MRS1000.cpp)
-target_link_libraries(MRS1000 LMS1xx ${console_bridge_LIBRARIES})
+target_link_libraries(MRS1000 CoLaA ${console_bridge_LIBRARIES})
 
 
 # Regular catkin package follows.
 find_package(catkin REQUIRED COMPONENTS roscpp sensor_msgs)
-catkin_package(CATKIN_DEPENDS roscpp)
+catkin_package(CATKIN_DEPENDS roscpp sensor_msgs)
+
+include_directories(include SYSTEM ${catkin_INCLUDE_DIRS})
```

Comment:
due to the reason explained above, here we should not add any `SYSTEM` tags, as in `${catkin_INCLUDE_DIRS}` are all packages we depend on. Those could be from an underlay or from the same workspace. Those warnings should then still pop up.

If those warnings are actually bad/annoying, one could consider fixing them upstream, even though this will probably only make it into lunar, or even m-turtle.

(One could argue that we'll never want `roscpp` or `sensor_msgs` in an underlay, but still, this kind of delivers a precedence...)